src/compressor/QatAccel.h

```diff
@@ -16,17 +16,26 @@
 #define CEPH_QATACCEL_H
 
 #include <memory>
+#include <mutex>
+#include <condition_variable>
+#include <vector>
+#include <atomic>
 #include <boost/optional.hpp>
 #include "include/buffer.h"
 
-extern "C" struct QzSession_S; //struct QzSession_S comes from QAT libraries
+extern "C" struct QzSession_S; // typedef struct QzSession_S QzSession_T;
 
 struct QzSessionDeleter {
   void operator() (struct QzSession_S *session);
 };
 
 class QatAccel {
-  std::unique_ptr<struct QzSession_S, QzSessionDeleter> session;
+  friend struct cached_session_t;
+  using session_ptr = std::unique_ptr<struct QzSession_S, QzSessionDeleter>;
+  static std::vector<session_ptr> sessions;
+  static std::mutex mutex;
+  static std::condition_variable cond;
+  static std::atomic<bool> init_called;
```

Comment:
why make this variable `atomic`?

Reply:
I'm used to using atomic types for variables that I think should be changed atomically, which have been protected with mutex and can be omitted. has been removed.